dali/kernels/transpose/transpose_gpu_impl.cuh

```diff
@@ -129,51 +127,55 @@ __device__ void TransposeTiledStatic(TiledTransposeDesc<T> desc) {
   pos[ndim - 1] = tile_x * kTileSize;
   pos[ndim - 2] = tile_y * kTileSize;
 
-  T *out = desc.out;
-  const T *in = desc.in;
+  T *out = reinterpret_cast<T *>(desc.out);
+  const T *in = reinterpret_cast<const T *>(desc.in);
 
   for (uint64_t tile = start_tile;;) {
     uint64_t in_ofs = 0, out_ofs = 0;
-    #pragma unroll
+#pragma unroll
     for (int d = 0; d < ndim - 2; d++) {
-      in_ofs  += desc.in_strides[d] * pos[d];
-      out_ofs += desc.out_strides[d] * pos[d];
+      in_ofs += desc.in_strides[d] / pack_ratio * pos[d];
+      out_ofs += desc.out_strides[d] / pack_ratio * pos[d];
```

Comment:
Shouldn't it be:
```suggestion
      in_ofs += (desc.in_strides[d] >> pack_ratio) * pos[d];
      out_ofs += (desc.out_strides[d] >> pack_ratio) * pos[d];
```
?

Reply:
repaired the lines, and added test for bigger number of dimensions, as I think current tests didn't cover this area of code (because there was dividing by 0)

Reply:
re-ran benchmarks and results didn't change